Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp

```diff
@@ -142,8 +142,10 @@ TrackPrivateBaseGStreamer::~TrackPrivateBaseGStreamer()
 
 void TrackPrivateBaseGStreamer::disconnect()
 {
-    if (m_stream)
+    if (m_stream) {
         g_signal_handlers_disconnect_matched(m_stream.get(), G_SIGNAL_MATCH_DATA, 0, 0, nullptr, nullptr, this);
+        ASSERT(0 == g_signal_handler_find(m_stream.get(), G_SIGNAL_MATCH_UNBLOCKED, 0, 0, nullptr, nullptr, nullptr));
```

Comment:
This assert doesn't make sense, because in its current form it's not matching by `this` data (it matches `nullptr`data). This means that if any other handler is registered to the stream for any other object (any other data), the assert will fail. This has created trouble upstream: https://bugs.webkit.org/show_bug.cgi?id=266899
I'm going to remove the assert.

Reply:
Hi @eocanha The idea was, by this time all the handlers should have been removed (including any specialization of TrackPrivateBaseGStreamer). The assert tries to ensure that.